src/olympia/promoted/admin.py

```diff
@@ -81,3 +109,9 @@ def formfield_for_foreignkey(self, db_field, request=None, **kwargs):
             return SlugOrPkChoiceField(**kwargs)
         return super().formfield_for_foreignkey(
             db_field, request, **kwargs)
+
+    def has_delete_permission(self, request, obj=None):
+        qs = PrimaryHeroInline.model.objects.filter(enabled=True)
+        if obj and list(qs) == [getattr(obj, 'primaryhero', None)]:
```

Comment:
Will this be true if there is only one enabled `primaryhero`, and it is the same as the current `primaryhero`? If so, couldn't the same thing be done by just counting the number of enabled `primaryhero`s if the current one is enabled, and checking that there is only one?

Reply:
Yes.  I did it this way in DiscoveryItemAdmin so I just copied it over, but I can change it if you want?

Reply:
It seems like the code will be a bit simpler with my suggestion. I had to stop and think about this one. Simpler is usually better, imo.

Reply:
https://github.com/mozilla/addons-server/pull/15078/commits/c3759260619cb9d23bded23584fb801b2834f92c